subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/ZipHasher.java

```diff
@@ -32,82 +34,99 @@
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nullable;
+import java.io.File;
 import java.io.IOException;
-import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Paths;
 import java.util.List;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipInputStream;
+import java.util.Set;
 
-public class JarHasher implements RegularFileHasher, ConfigurableNormalizer {
-    private static final Logger LOGGER = LoggerFactory.getLogger(JarHasher.class);
+public class ZipHasher implements RegularFileHasher, ConfigurableNormalizer {
 
-    private final ResourceHasher classpathResourceHasher;
-    private final ResourceFilter classpathResourceFilter;
+    private static final Set<String> KNOWN_ZIP_EXTENSIONS = ImmutableSet.of("zip", "jar", "war", "ear", "apk");
```

Comment:
As explained below, I think these should probably include the leading `.`, if you'll continue using `hasExtensionIgnoresCase`